sdk/clientcore/core/src/main/java/io/clientcore/core/util/Union.java

```diff
@@ -80,7 +79,9 @@ public final class Union {
     private Type currentType;
 
     private Union(Type... types) {
-        this.types = Arrays.asList(types);
+        ArrayList<Type> typeCopy = new ArrayList<>(types.length);
+        Collections.addAll(typeCopy, types);
+        this.types = typeCopy;
     }
```

Comment:
Something like this may suffice:

```java
private Union(Type... types) {
    this.types = Arrays.asList(Arrays.copyOf(types, types.length));
}
```

Reply:
Same reason as in https://github.com/Azure/azure-sdk-for-java/pull/43778#discussion_r1927644075